runelite-client/src/main/java/net/runelite/client/plugins/puzzlesolver/solver/pathfinding/IDAStarMM.java

```diff
@@ -0,0 +1,791 @@
+						if (distY >= 2)
+						{
+							Point locSwap = new Point(locBlank.getX(), locBlank.getY() + 1);
+							swap(locBlank, locSwap);
+						}
+						else if (distY <= -2)
+						{
+							Point locSwap = new Point(locBlank.getX(), locBlank.getY() - 1);
+							swap(locBlank, locSwap);
+						}
+					}
+				}
+				else if (distY == 0)
+				{
+					//Same row
+					if (distAbsX == 1)
+					{
+						//Next to
+						reached = true;
+					}
+					else
+					{
+						//More than 2 away, move towards on X-axis
+						if (distX >= 2)
+						{
+							Point locSwap = new Point(locBlank.getX() + 1, locBlank.getY());
+							swap(locBlank, locSwap);
+						}
+						else if (distX <= -2)
+						{
+							Point locSwap = new Point(locBlank.getX() - 1, locBlank.getY());
+							swap(locBlank, locSwap);
+						}
+					}
+				}
+				else
+				{
+					//Different row and column
+					if (rowMode)
+					{
+						//Check if already correct above
+						if (locBlank.getY() - 1 == row &&
+							validRowNumbers.get(row).contains(currentState.getPiece(locBlank.getX(), locBlank.getY() - 1)) &&
+							currentState.getPiece(locBlank.getX(), locBlank.getY() - 1) < valTarget &&
+							distY <= -1)
+						{
+							//Move forward
+							Point locSwap = new Point(locBlank.getX() + 1, locBlank.getY());
+							swap(locBlank, locSwap);
+							continue;
+						}
+
+						//Move upwards or downwards
+						if (distY >= 1)
+						{
+							Point locSwap = new Point(locBlank.getX(), locBlank.getY() + 1);
+							swap(locBlank, locSwap);
+						}
+						else if (distY <= -1)
+						{
+							Point locSwap = new Point(locBlank.getX(), locBlank.getY() - 1);
+							swap(locBlank, locSwap);
+						}
+					}
+					else
+					{
+						//Check if already correct to the left
+						if (locBlank.getX() - 1 == column &&
+							validColumnNumbers.get(column).contains(currentState.getPiece(locBlank.getX() - 1, locBlank.getY())) &&
+							currentState.getPiece(locBlank.getX() - 1, locBlank.getY()) < valTarget &&
+							distX <= -1)
+						{
+							//Move down
+							Point locSwap = new Point(locBlank.getX(), locBlank.getY() + 1);
+							swap(locBlank, locSwap);
+							continue;
+						}
+
+						//Move left or right
+						if (distX >= 1)
+						{
+							Point locSwap = new Point(locBlank.getX() + 1, locBlank.getY());
+							swap(locBlank, locSwap);
+						}
+						else if (distX <= -1)
+						{
+							Point locSwap = new Point(locBlank.getX() - 1, locBlank.getY());
+							swap(locBlank, locSwap);
+						}
+					}
+				}
+			}
+			else
+			{
+				//Swap towards locTarget
+				if (rowMode)
+				{
+					alignTargetX(valTarget, locTarget);
+					swapUpRow(valTarget, locTarget);
+				}
+				else
+				{
+					alignTargetY(valTarget, locTarget);
+					swapLeftColumn(valTarget, locTarget);
+				}
+			}
+		}
+	}
+
+	private void alignTargetX(int valTarget, Point locTarget)
+	{
+		Point locVal = findPiece(valTarget);
+		Point locBlank;
```

Comment:
This isn't C, consider declaring ``locBlank`` the first time it's used.
That makes it easier to see in which scope the variable is used.
It's overwritten every iteration so in this case it shouldn't be a problem?

Reply:
Yeah, moved it